### PR TITLE
Congrats pages: Refactor Titan page

### PR DESCRIPTION
**Pull request by nightnei**
Related to https://github.com/Automattic/dotcom-forge/issues/2729

## Proposed Changes
This is a redesign of Titan-related congrats pages, as part of the project to redesign all congrats pages (pb5gDS-38y-p2).

###  [Set up mailbox: `controller.js` -> `emailManagementTitanSetUpThankYou`](#email-management-titan-set-up-thank-you)
1. Buy Titan
2. Open Upgrades -> Emails -> Select Titan subscription from previous step
3. Remove all mailboxes from the subscription
4. You should see `Set up mailbox`, click on it and complete set up of the mailbox
7. You should be landed to the next page: `http://calypso.localhost:3000/email/<DOMAINNAME>/titan/set-up-mailbox/thank-you/<MAIN_DOMAIN>?email=<MAILBOX>`

Screenshot 2024-02-02 at 13 10 13

### [Buy titan subscription w/o mailboxes: `selectedFeature === 'email-license' && domainName'](#titan-wo-mailboxes)
1. Open `http://calypso.localhost:3000/start/onboarding-with-email?ref=professional-email` (Emulates click on "Start your free" trial on https://wordpress.com/professional-email/)
2. Proceed with buying domain and adding/buying Titan subscription (We buy only subscription, w/o mailboxes)
3. You should be landed to the next page: `http://calypso.localhost:3000/checkout/thank-you/features/email-license/<MAIN_DOMAIN_NAME>/<SUBSCRIPTION_ID>?siteId=<SITE_ID>&notice=purchase-success`

Screenshot 2024-02-08 at 17 04 36

### Standart way of buying Titan + 1 mailbox
1. Buy a domain w/o email
1. Open Upgrades -> Emails
2. Click "Add email"
3. Fill in Titan mailbox fields (only 1, later we will tests a few, there is another data on the screen)
4. Click "Purchase"
5. Complete checkout
6. You should be landed to the next page: `http://calypso.localhost:3000/checkout/thank-you/<MAIN_DOMAIN_NAME>/<SUBSCRIPTION_ID>?email=<MAILBOX>`

Screenshot 2024-02-02 at 16 05 26

### Buying additionally a few mailboxes
1. We will continue working with what we got from `Standart way of buying Titan + 1 mailbox` testing instructions
2. Open `Upgrades` -> `Emails`
3. Select the subscription from previous tet instruction `Standart way of buying Titan + 1 mailbox`
4. Click on "Add new mailboxes"
5. Fill in fields from 2+ mailboxes
6. Proceed with checkout
7. You should be landed to the next page: `http://calypso.localhost:3000/checkout/thank-you/<MAIN_DOMAIN_NAME>/<SUBSCRIPTION_ID>?email=<MAILBOX>`
8. Assert that in the product list you have only one item, but there is written the number of mailboxes, e.g. `3 mailboxes for nightnei-a-few-mb.blog`

Screenshot 2024-02-12 at 19 30 31

### CODE REVIEW

File: client/components/thank-you-v2/footer/index.tsx
Comment (nightnei) [inline client/components/thank-you-v2/footer/index.tsx:26]: We have case when we have only `href`, w/o buttonOnClick. And IMO it's totally make sense - sometimes can be provided only one of them and it should be visible.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11292969

Thank you @nightnei for including a screenshot in the description! This is really helpful for our translators.

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (stephanethomas) [inline client/my-sites/checkout/checkout-thank-you/index.tsx:111]: Could we order those imports alphabetically so they are easier to read?

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/email-product.tsx
Comment (stephanethomas) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/email-product.tsx]: Since this component is only for Titan, this should appear in its name - e.g. `ThankYouTitanProduct`. Another approach would be to rework that component to also support Google Workspace (by the way, it this work planned? I'd like to make sure we address all email providers).

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/utils.ts
Comment (stephanethomas) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/utils.ts]:
```suggestion
import {
	getDomainPurchase,
	isBulkDomainTransfer,
	isDomainOnly,
	isPurchaseTitanWithoutMailboxes,
} from '../utils';
```

File: client/my-sites/checkout/checkout-thank-you/utils.ts
Comment (stephanethomas) [inline client/my-sites/checkout/checkout-thank-you/utils.ts]: That's minor but I feel `isTitanWithoutMailboxesPurchase` reads better in English. Note though we could probably remove that mention entirely, since a) it acts on a _feature_ and not a _purchase_, and b) even functions such as `isDomainRegistration` don't include it in their name.

**Review by stephanethomas — APPROVED** [review]
This works great, except maybe for one tiny tiny thing - there is a white background on those two links for the first flow from the testing instructions:

image

Other than that, my main concern is that the `ThankYouEmailProduct` is tailored for Titan whereas this is not reflected in its name. I'm also curious about the plans for Google Workspace:

image

This is not a blocker though, so I'm going to accept that pull request 🙂.

File: client/my-sites/checkout/checkout-thank-you/utils.ts
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/utils.ts]: I wanted to make condition more readable, but yeah, probably it's better to avoid using extra function

Comment (stephanethomas) [reply client/my-sites/checkout/checkout-thank-you/utils.ts]: Oh I would keep this function! I was only talking about the `Purchase` mention 🙂.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/email-product.tsx
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/email-product.tsx]: > Since this component is only for Titan, this should appear in its name - e.g. ThankYouTitanProduct. 

Good point

>Another approach would be to rework that component to also support Google Workspace 

I was thinking about it, but the content to the GW page is different, that's why I decided to make another component for GW later.

>by the way, it this work planned? I'd like to make sure we address all email providers

Yes, I am going to do it right after `Titan Congrats` and `Developer Survey`. https://github.com/orgs/Automattic/projects/865/views/1?pane=issue&itemId=49015632

File: client/my-sites/checkout/checkout-thank-you/utils.ts
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/utils.ts]: Got it, reverted and changed the name :)

**Comment by nightnei** [comment]
@stephanethomas Also addressed white background for links

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/index.tsx]:
```suggestion
				// Users may purchase Titan subscription without specifying the mailbox name using
				// the onboard with email flow (https://wordpress.com/start/onboarding-with-email/mailbox-domain)
```

I think it's clearer to describe the scenario rather than pointing to the PR.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx:21]:
```suggestion
export function ThankYouEmailProduct( {
```
I would consider creating a generic component for email products (Titan / Google Workspace), since they share some similar logic. Although, I do realize that _some_ logic needs to be customized for each provider, e.g. product name, inbox URL, etc. I'll leave it for you to decide!

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]: I just realize this doesn't quite work for all languages. Here, we're assuming that the product name and details are read like this: "Professional Email for {domainName}". While that works for English, it may not necessarily work for other languages.

I also just realized that we only show the "Set up mailbox" button in the current Congrats Page. So I'd like to propose the following:

Screen Shot 2024-02-08 at 9 14 33 PM

**Title:** Congratulations on your purchase! (keep the same title as the current Congrats page)
**Subtext:** Complete your professional email setup to start sending and receiving emails from your custom domain today.
**Product name:** Professional Email
**Product detail:** Set up for {domain-name}
**Product CTA:** Set up mailbox (we won't have a header button)

The reason why I suggested using the old title "Congratulations on your purchase!" is to avoid repeating "Set up your email" over and over again.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]:
```suggestion
			name={ translate( 'Professional Email' ) }
```
"Professional Email" is a name, so we use title case. See headline in this landing page: https://wordpress.com/professional-email/

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]: I recall we discussed about handling multiple mailboxes in this on Slack. Ideally, we want to render a product item per mailbox. However, we decided _not_ to proceed with that approach, since the current implementation of generating the thank you URL only passes one email address in the query string parameter, even when multiple mailboxes are purchased. To support render multiple product items, we will need to update the thank you URL logic, which is out of scope of this refactoring task.

That said, we can still improve UX by providing clearer details in the product item. Instead of rendering the email address passed from the query string, we can render: `{num} mailboxes for {domainName}`:

Screen Shot 2024-02-08 at 9 37 33 PM

The "Go to inbox" link will still be generated based on the first email address, but given that's the existing behaviour in production, I think it's fine to keep the same logic for now.

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/index.tsx]:
```suggestion
			} else if ( isTitanWithoutMailboxes( selectedFeature ) ) {
				// Users may purchase Titan subscription without specifying the mailbox name using
				// the onboard with email flow (https://wordpress.com/start/onboarding-with-email/mailbox-domain)
				const titanPurchase = purchases.find( ( purchase ) => isTitanMail( purchase ) );
			
				pageContent = titanPurchase && (
					<TitanSetUpThankYou
						domainName={ titanPurchase.meta }
						isDomainOnlySite={ this.props.domainOnlySiteFlow }
					/>
				);
```

The suggested revisions address two things:
* I think it's clearer to describe the scenario rather than pointing to a PR.
* We can get the domain from the titan purchase instead of getting the domain purchase.

File: client/my-sites/email/controller.js
Comment (wongasy) [inline client/my-sites/email/controller.js]: Similarly, I would avoid pointing to a PR. If we are looking to provide context of this feature, it's best to describe the scenario rather than pointing to the PR and refer to test steps.

```suggestion
	// Thank you page for when user sets up a mailbox after purchasing a Titan subscription.
```

After going through the flow again, I think it should be changed altogether. The Congrats Page is meant to be shown in the "checkout experience" (without sidebar, without the admin bar). We can still proceed with as-is in this PR, so that the old Congrats page is removed. But I would suggest creating a followup task to:

* Redirect users back to the Email Management page after setting up a mailbox (no payment was involved, the action was really part of management, it doesn't make sense for us to show the Congrats page)
* Display a notice "Your email is now ready to use!" (copy taken from old Congrats page).

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/utils.ts
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/utils.ts:34]: Similarly to the earlier comment in `<CheckoutThankYou>` component. We probably don't need to get the domain purchase, instead, I would check for titan purchase.

**Review by wongasy — CHANGES_REQUESTED** [review]
Looking good! I've left a few feedback.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx:21]: `Titan` was used intentionally and `roduct name, inbox URL, etc` were main reasons. The component now is small and straightforward, if we mix Titan and GW - we will have it more complicated with a lot of conditions. So I would create 2 different component.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]: Regarding this - I would keep it as is, my concerns:
1) `Go to inbox` looks unpredictable for users, since it's not clear which inbox will be opened, 1-st or 2nd.
2) Both options have some downsides, but with the current option our users have lived for years, so I would vote fir keeping it as is, and later we will fix it totally in good way.

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/index.tsx]: IMO the current approach a bit more clear, let me elaborate:
1) I know that the flow is buying domain and titan, so the condition about "buying domain" is more clear, when I see only the condition about "titan purchase". As "domain and titan w/o mailboxes", instead of "titan purchase and titan w/o mailboxes".
2) Just personal - it's more clear at a glance when we have one condition, instead of two, and the second one is located inside the first condition, but it's just personal.
3) `domainName={ domainPurchase.meta }` looks a bit more readable and clear, than `domainName={ titanPurchase.meta }`.

All my arguments are obviously and not strong, so if you like much more `titanPurchase `, I am totally ok to go with it. Waiting for your final decision, taking into account my points above.

File: client/my-sites/email/controller.js
Comment (nightnei) [reply client/my-sites/email/controller.js]: 👍 Created https://github.com/orgs/Automattic/projects/865/views/1?pane=issue&itemId=52788499

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (wongasy) [reply client/my-sites/checkout/checkout-thank-you/index.tsx]: >I know that the flow is buying domain and titan, so the condition about "buying domain" is more clear, when I see only the condition about "titan purchase". As "domain and titan w/o mailboxes", instead of "titan purchase and titan w/o mailboxes".

I can agree with this. It's true that in our current set up, the only way users would end up purchasing a Titan subscription without defining a name for the mailbox is when they purchase a domain AND email subscription. If you prefer keeping `domainPurchase`, that's fine with me! But I would still suggest cleaning up the comment so we describe the scenario rather than pointing to a PR :)

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (wongasy) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]: The existing implementation in production looks like this:

Screen Shot 2024-02-09 at 8 41 21 AM

Note that we don't show the email address. I suspect rendering only ONE of the email addresses can be confusing, as the user bought more than one. So I would suggest specifying that in the details as an improvement.

Perhaps we could stick with "Your Mailboxes" (what we have in production) instead of "Go to inbox".

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/index.tsx]: >But I would still suggest cleaning up the comment so we describe the scenario rather than pointing to a PR :)

Yeah, I have already changed both comments :)

File: client/my-sites/email/controller.js
Comment (wongasy) [reply client/my-sites/email/controller.js]: Thanks for creating the issue! The description doesn't look exactly accurate though, I'll clarify in the issue itself.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]: I faced one blocker - [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/email/controller.js#L193) we have information only from url, so we don't know the number of mailboxes.

Comment (wongasy) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]: I see the quantity listed in the purchase item of the `receipt` endpoint. There should be a way to retrieve that value. Perhaps try the earlier suggested change https://github.com/Automattic/wp-calypso/pull/87098#discussion_r1483813040?

Screen Shot 2024-02-09 at 11 39 45 AM

Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]: It's another place, when we purchased only mailbox - https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/email/controller.js?rgh-link-date=2024-02-09T16%3A36%3A12Z#L193

File: client/my-sites/email/titan-set-up-thank-you/index.tsx
Comment (emilyaudela) [inline client/my-sites/email/titan-set-up-thank-you/index.tsx]: Should the `f` in `find answers` be capitalized? IMO, the phrasing here is a little weird. Because "find" is a verb, you kind of assume "email" would be a verb too, but here, it's referring to "questions about Email." Maybe it would be better if it wasn't a question, but something simpler like "Find answers to your questions about Email"

Comment (nightnei) [reply client/my-sites/email/titan-set-up-thank-you/index.tsx]: Yeah, IMO your suggestion looks better 👍
@wongasy do you agree with the suggestion?

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]: Btw, I have thought - I think that it's better to do this UI change together with this task: https://github.com/Automattic/wp-calypso/issues/87334
Since we will remove the route behind `controller.js`, and will be using only one place for rendering both cases.
WDYT?

File: client/my-sites/email/titan-set-up-thank-you/index.tsx
Comment (wongasy) [reply client/my-sites/email/titan-set-up-thank-you/index.tsx]: @nightnei I think we should follow what was outlined in the copy document instead of figma. This is what we have in the doc:
> Email questions? We have the answers

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]: Closing the loop (after discussion in Slack) - we are rendering the number of mailboxes always when we have such information, other wise we render email address from URL-query.

Since `Set up mailbox` leads to page with sidebar where we doesn't retrieve information from backend, we have only query parameters.

Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]: Now we have the quantity of purchased mailboxes and added the steps to reproduce it into testing instructions in the PR's description.

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/index.tsx]:
```suggestion
						numberOfMailboxesPurchased={ titanPurchase?.newQuantity }
```
I realize that the prop name was meant to match the endpoint property, but it doesn't quite provide a clear description of what the prop represents. I'd suggest renaming the prop to `numberOfMailboxesPurchased`.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx]:
```suggestion
		details = translate( 'Set up your mailbox for %(domainName)s', { args: { domainName } } );
```
Sorry, I might have missed the words "your mailbox" when I suggested the change.

File: client/my-sites/email/titan-set-up-thank-you/index.tsx
Comment (wongasy) [inline client/my-sites/email/titan-set-up-thank-you/index.tsx]: I guess the idea here is to cover the case where user purchased Titan without specifying a mailbox name, which is currently described as `isTitanWithoutMailboxes()`. That isn't exactly accurate though, the user did purchase a mailbox, it's just that the mailbox is not yet set up. To avoid this confusion, I would suggest:

* Renaming the function `isTitanWithoutMailboxes()` to `isTitanPurchaseWithoutMailboxName()`.
* Make `newQuantityOfMailboxes` (I'd suggest renaming to `numberOfMailboxesPurchased`) an optional prop instead of falling back to `0`, since users will never purchase 0 mailbox.
*  In `<ThankYouTitanProduct>`, the `if/else` conditions become:

```
if ( newQuantityOfMailboxes && newQuantityOfMailboxes > 1 ) {
    // Render number of mailboxes purchased
} else if ( ! emailAddress ) {
    // Show messaging asking user to set up the mailbox
} else {
    // Render mailbox name
}
```

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/titan-product.tsx:48]:
```suggestion
				target="_blank"
				rel="noopener noreferrer"
				onClick={ () => {
```
Let's ensure we open Titan webmail in a new tab, since we're taking users away from WordPress.com.

**Review by wongasy — APPROVED** [review]
LGTM with a few comments that we should address. I guess we'd also need to wait for translation to be completed before deploying.

The Congrats page rendered properly for all scenarios:

* Purchased domain and Professional Email using the "onboard with email" flow, where users are not required to specify the mailbox name.

Screen Shot 2024-02-12 at 7 48 59 PM

* Purchased Professional Email (one mailbox) via the Upgrades > Email flow.

Screen Shot 2024-02-12 at 7 40 45 PM

* Purchased Professional Email (multiple mailboxes) via the Upgrades > Email flow

Screen Shot 2024-02-12 at 7 32 30 PM

* After setting up a previously purchased mailbox (Note that this screen will eventually be removed. We'd simply redirect users back to the Email Management page after successfully setting up a mailbox).

Screen Shot 2024-02-12 at 7 42 03 PM

File: client/my-sites/email/titan-set-up-thank-you/index.tsx
Comment (nightnei) [reply client/my-sites/email/titan-set-up-thank-you/index.tsx]: Adjusted everything, except one thing - as for me, `isTitanWithoutMailboxes` looks a bit better.
Since:
1) Can be 1 or a few mailboxes, so it's better to use plural `Mailboxes` instead of singular `MailboxName`
2) `isTitan` vs `isTitanPurchase` - it was discussed here (https://github.com/Automattic/wp-calypso/pull/87098#discussion_r1482794459) with Stephane, we ended up to not add it.

**Comment by fredrikekelund** [comment]
Sorry for the drive-by observation, but per our old discussion about PR titles, could we stick to the plain English format in Calypso too? 😊

So, in this case, something like `Congrats pages: Refactor Titan page` or even just `Refactor Titan thank you page`

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11292969).

**Pull request merged**
